Source/CarthageKit/Cartfile.swift

```diff
@@ -92,6 +92,35 @@ extension Cartfile: Printable {
 	}
 }
 
+// Duplicates
+// TODO: use Set when Carthage supports Swift 1.2
+extension Cartfile {
+	/// Returns an array containing projects that are listed as duplicate
+	/// dependencies.
+	public func duplicateProjects() -> [ProjectIdentifier] {
+		return repeatedElements(self.sortedProjects())
+	}
+
+	/// Returns an array containing projects that are listed as dependencies
+	/// in both the receiver and the argument.
+	public func duplicateProjects(cartfile: Cartfile) -> [ProjectIdentifier] {
```

Comment:
This feels more like a static method or free function to me (since the receiver and the argument conceptually have equal weight)—perhaps `duplicateProjectsInCartfiles` or something.